### PR TITLE
A few AsyncSeq operations

### PR DESCRIPTION
**Pull request by eulerfx**
- toArray, toList, unfoldAsync, concatSeq, interleave, bufferByCount 
- also added Async.map, Async.bind, Async.unit

### CODE REVIEW

File: src/FSharpx.Async/Async.fs [hunk 1]
```diff
@@ -8,10 +8,14 @@ open System.Threading
 
 // ----------------------------------------------------------------------------
 
-[<AutoOpen>]
-module AsyncExtensions = 
-    type Microsoft.FSharp.Control.Async with 
+module AsyncOps =
```
Comment (tpetricek) [inline src/FSharpx.Async/Async.fs:11]: Should this be `internal`?

**Comment by tpetricek** [comment]
Looks good!

File: src/FSharpx.Async/Async.fs (hunk 1 above)
Comment (eulerfx) [inline src/FSharpx.Async/Async.fs:11]: The only issue with making it internal is that then you can't inline the methods on `Async` that would call it. Premature optimization?

**Comment by eulerfx** [comment]
@forki @tpetricek  Is there anything I should change in this PR?

**Pull request merged**